src/utils/asPromise.ts

```diff
@@ -0,0 +1,7 @@
+export const asPromise = (val: any): Promise<any> => {
```

Comment:
Might as well use generic type

(have not tested)

```suggestion
export const asPromise = <T>(val: T): Promise<T> => {
```